File: Plugins/Aspose_Tasks_Java_for_Ruby/lib/asposetasksjava/ResourceAssignments/createresourceassignment.rb
module Asposetasksjava
  module CreateResourceAssignment
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'
                
        # Instantiate project object
        project = Rjb::import('com.aspose.tasks.Project').new(data_dir + 'test_tasks.mpp')

        task = project.getRootTask().getChildren().add("Task")

    	rsc = project.getResources().add("Rsc")

    	assn = project.getResourceAssignments().add(task, rsc)

    	puts "Created resource assignment."
    end
  end
end

File: Plugins/Aspose_Tasks_Java_for_PHP/src/aspose/tasks/WorkingWithResourceAssignments/CreateResourceAssignment.php
<?php
namespace Aspose\Tasks\WorkingWithResourceAssignments;

use com\aspose\tasks\Project as Project;
class CreateResourceAssignment{

    public static function run($dataDir=null){

        # Instantiate project object
        $project = new Project($dataDir . 'test_tasks.mpp');

        $task = $project->getRootTask()->getChildren()->add("Task");

    	$rsc = $project->getResources()->add("Rsc");

    	$assn = $project->getResourceAssignments()->add($task, $rsc);

    	print "Created resource assignment.".PHP_EOL;

    }

}
?>